src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -16,48 +19,135 @@ export interface BreadcrumbItem {
 export interface BreadcrumbProps extends Pick<LinkProps, 'as'> {
   /** A list of `BreadcrumbsItem` objects ( `{href,children}` ) that will construct the Breadcrumb */
   items: BreadcrumbItem[];
+  /** Truncates component if breadcrumb list is too long.
+   * At least 4 breadcrumbs must exist for truncation to occur.
+   */
+  truncate?: boolean;
 }
 
+interface IBreadcrumbLink {
+  crumb: BreadcrumbItem;
+  isLastBreadcrumb?: boolean;
+  showChevron?: boolean;
+}
+
+type ITruncatedBreadcrumbs = Omit<BreadcrumbProps, 'truncate'>;
+
+const BreadcrumbChevron: React.ElementType = () => {
+  return <Icon type="chevron-right" mx={2} color="blue-300" size="medium" role="presentation" />;
+};
+
+const BreadcrumbLink: React.ElementType = ({
+  crumb,
+  isLastBreadcrumb = false,
+  showChevron = true,
+  ...rest
+}: IBreadcrumbLink) => {
+  return (
+    <Flex
+      key={crumb.href}
+      as="li"
+      alignItems="center"
+      fontSize="medium"
+      fontWeight={isLastBreadcrumb ? 'bold' : 'normal'}
+      aria-current={isLastBreadcrumb ? 'page' : undefined}
+    >
+      <Link
+        to={crumb.href}
+        variant="neutral"
+        border="none"
+        truncated
+        maxWidth="450px"
+        data-active={isLastBreadcrumb ? true : undefined}
+        {...rest}
+      >
+        {crumb.children}
+      </Link>
+      {!isLastBreadcrumb && showChevron && <BreadcrumbChevron />}
+    </Flex>
+  );
+};
+
+const TruncatedBreadcrumbs: React.ElementType = ({ items, ...rest }: ITruncatedBreadcrumbs) => {
+  const firstItem = items[0];
+  const truncatedList = items.slice(1, -2);
+  const remainingCrumbs = items.slice(-2);
+
+  return (
+    <React.Fragment>
+      {/* first crumb */}
+      <BreadcrumbLink crumb={firstItem} {...rest} />
+
+      {/* truncated crumbs */}
+      <Popover>
+        {() => (
+          <Flex alignItems="center">
+            <PopoverTrigger
+              as={Button}
+              id="truncated-breadcrumb-list"
```

Comment:
nit: do we need this ID? If we don't use it I'd remove it just in case there's a duplicate element in the same page in another component